app/bundles/LeadBundle/Entity/DoNotContactRepository.php

```diff
@@ -137,10 +137,18 @@ public function getTimelineStats($leadId = null, array $options = [])
     }
 
     /**
-     * @param array $contacts Array of contacts to filter by
+     * @param string|null    $channel
+     * @param string[]|int[] $contacts Array of contact IDs to filter by
+     *
+     * @return mixed[]
      */
     public function getChannelList($channel, array $contacts = null): array
     {
+        // If no contacts are sent then stop querying for all of the DNC records as it leads to the out of memory error.
+        if (is_array($contacts) && empty($contacts)) {
```

Comment:
```suggestion
        if (empty($contacts)) {
```

IMO, we should not use is_array check here as empty() will take care for both (null and [])

right now if value of $contacts is null then also queries should not be execute but as per if condition query will execute.

Reply:
or define default value of $contacts is [] instead of null

Reply:
It kinda seem like the is_array() function was there for some reason but I can't figure it out nor remember as I was working on this 2 years ago. Let's merge your suggestion.

Reply:
@dadarya0 it started failing tests. It seems that the is_array was there to avoid a BC break. I'll revert the last commit.